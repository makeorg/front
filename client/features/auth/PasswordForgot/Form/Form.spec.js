import React from 'react';
import renderer from 'react-test-renderer';
import { ForgotPasswordFormComponent } from './index';

describe('ForgotPasswordFormComponent', () => {
  it('must match the snapshot by default', () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer.create(
      <ForgotPasswordFormComponent errors={[]} />
    );
    expect(component).toMatchSnapshot();
  });

  it('must match the snapshot with errors', () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer.create(
      <ForgotPasswordFormComponent errors={[{
        field: 'email',
        message: 'Error message 1'
      }, {
        field: 'field',
        message: 'Error message 2'
      }]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
