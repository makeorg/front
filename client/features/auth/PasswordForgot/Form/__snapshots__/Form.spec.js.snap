// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordFormComponent must match the snapshot by default 1`] = `
.c4 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(232,24,68);
}

.c5 {
  margin-right: 0.375rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 1.125rem;
  border-radius: 1.875rem;
  border: 0.0625rem solid rgba(0,0,0,0.2);
  margin-bottom: 0.9375rem;
  background-color: rgb(242,242,242);
  border-color: rgba(0,0,0,0.2);
}

.c3 {
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 0.625rem;
  width: 100%;
  color: rgba(0,0,0,0.55);
  font-size: 0.875rem;
  line-height: 2.188rem;
}

.c2 {
  width: 1.875rem;
  color: rgb(232,24,68);
  font-size: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: 0.625rem 0 0;
  max-width: 30.63rem;
}

@media (min-width:47.94rem) {
  .c4 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c5 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c3 {
    font-size: 1rem;
    line-height: 2.5rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    font-size: 1.125rem;
  }
}

<form
  className="c0"
  id="forgot_password"
>
  <div
    className="c1"
  >
    <label
      aria-label="common.form.email_label"
      className="c2"
      htmlFor="email"
    >
      <svg
        icon="envelope"
        lib="@fortawesome/react-fontawesome"
        prefix="far"
      />
    </label>
    <input
      aria-required={true}
      className="c3"
      id="email"
      name="email"
      placeholder="common.form.email_label"
      required={true}
      tabIndex={-1}
      type="email"
    />
  </div>
  <button
    className="c4"
    form="forgot_password"
    tabIndex={-1}
    type="submit"
  >
    <span
      className="c5"
    >
      <svg
        icon="paper-plane"
        lib="@fortawesome/react-fontawesome"
        prefix="far"
      />
    </span>
    forgot_password.send_link
  </button>
</form>
`;

exports[`ForgotPasswordFormComponent must match the snapshot with errors 1`] = `
.c5 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(232,24,68);
}

.c6 {
  margin-right: 0.375rem;
}

.c4 {
  color: rgb(237,24,68);
  margin-bottom: 0.625rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 1.125rem;
  border-radius: 1.875rem;
  border: 0.0625rem solid rgba(0,0,0,0.2);
  margin-bottom: 0.9375rem;
  background-color: rgb(242,242,242);
  border-color: rgb(237,24,68);
}

.c3 {
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 0.625rem;
  width: 100%;
  color: rgba(0,0,0,0.55);
  font-size: 0.875rem;
  line-height: 2.188rem;
}

.c2 {
  width: 1.875rem;
  color: rgb(232,24,68);
  font-size: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: 0.625rem 0 0;
  max-width: 30.63rem;
}

@media (min-width:47.94rem) {
  .c5 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c6 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c3 {
    font-size: 1rem;
    line-height: 2.5rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    font-size: 1.125rem;
  }
}

<form
  className="c0"
  id="forgot_password"
>
  <div
    className="c1"
  >
    <label
      aria-label="common.form.email_label"
      className="c2"
      htmlFor="email"
    >
      <svg
        icon="envelope"
        lib="@fortawesome/react-fontawesome"
        prefix="far"
      />
    </label>
    <input
      aria-required={true}
      className="c3"
      id="email"
      name="email"
      placeholder="common.form.email_label"
      required={true}
      tabIndex={-1}
      type="email"
    />
  </div>
  <p
    className="c4"
    id="authentification-email-error"
  >
    common.form.Error message 1
  </p>
  <button
    className="c5"
    form="forgot_password"
    tabIndex={-1}
    type="submit"
  >
    <span
      className="c6"
    >
      <svg
        icon="paper-plane"
        lib="@fortawesome/react-fontawesome"
        prefix="far"
      />
    </span>
    forgot_password.send_link
  </button>
</form>
`;
